python-package/xgboost/core.py

```diff
@@ -790,7 +807,7 @@ def set_uint_info(self, field: str, data) -> None:
         from .data import dispatch_meta_backend
         dispatch_meta_backend(self, data, field, 'uint32')
 
-    def save_binary(self, fname, silent=True) -> None:
+    def save_binary(self, fname: str | os.PathLike, silent: bool = True) -> None:
```

Comment:
Consider using Union instead.  This requires `from future import __annotations__`, which has different behavior.

Reply:
I'm not sure if that different behavior matters (except for the explicit import), but since `typing.Union` is available for Python <3.10, I agree it's better to stick with that.